siliconcompiler/core.py

```diff
@@ -787,6 +787,15 @@ def get(self, *keypath, field='value', job=None, cfg=None):
 
         """
 
+        # Ensure that all keypath values are strings.
+        # Scripts may accidentally pass in [None] if a prior schema entry was unexpectedly empty.
+        for kp in keypath:
+            if not type(kp) == str:
```

Comment:
A couple nits:
- I think [`isinstance()`](https://docs.python.org/3/library/functions.html#isinstance) is a bit more idiomatic than `type(...) == ...`
- `key` might be a better index variable name than `kp`

Reply:
Good idea, thanks.